### ISSUE TITLE
class <class> was compiled by a pre-release version of Kotlin and cannot be loaded by this version of the compiler.

### ISSUE THREAD

**Issue by realdadfish** [detekt/detekt#8639]
This is with 2.0.0-alpha.0. When Detekt runs with `debug = true`, hundreds of instances of the above warning are issued for my module. I only saw these because Detekt told me previously

```
There were 158 compiler errors found during analysis. This affects accuracy of reporting.
```

None of the classes it tells me however are in any kind special or are even part of an external library, they're all part of one big monorepo. I searched around and only found https://youtrack.jetbrains.com/issue/KT-75588 as reference, but this talks about issues with annotation-site targets. None of my classes have any annotations.

**Comment by 3flex** [comment]
There's not enough information here to investigate.

In the linked issue it says using unstable features can result in binaries marked as compiled with a pre-release version of Kotlin. Are you enabling any others? Have you set language version to 2.3?

Without a reproducer or more information I'm just guessing.

**Comment by realdadfish** [comment]
These are my compiler options:

```kotlin
compilerOptions {
    languageVersion.set(KotlinVersion.KOTLIN_2_2)
    progressiveMode.set(true)
    allWarningsAsErrors.set(true)
    extraWarnings.set(true)
    freeCompilerArgs.addAll(
        "-Xjsr305=strict",
        "-Xsuppress-version-warnings",
        "-Xwarning-level=UNUSED_ANONYMOUS_PARAMETER:disabled",
        "-Xwarning-level=REDUNDANT_VISIBILITY_MODIFIER:disabled",
        "-Xannotation-default-target=param-property",
        "-Xcontext-parameters",
        "-Xcontext-sensitive-resolution",
        "-Xannotation-target-all",
        "-Xnested-type-aliases",
        "-Xannotations-in-metadata"
    )
    optIn.addAll(listOf(
        "kotlin.time.ExperimentalTime",
        "kotlin.uuid.ExperimentalUuidApi"
    ))
}
```

**Comment by realdadfish** [comment]
Interestingly, none of the new features that are enabled are used by any of the offending classes. Some are really simple data classes.

**Comment by 3flex** [comment]
I don't know what the compiler does in this case but again I refer to the issue you linked from the top, where they that using an unstable feature can cause binaries to be marked as pre-release. This is likely the cause based on flags like `-Xcontext-sensitive-resolution` and `-Xnested-type-aliases`.

If you can make a smaller reproducer and share it will be much easier to investigate what's going on, and whether this is something we can work around or not. But given this is likely due to the use of experimental Kotlin features it's going to be lower on the priority list.

**Comment by realdadfish** [comment]
Ok, question here is whether this results into the non-analysis of all the offended files. Because if so I'd rather deactivate the experimental Kotlin features or try some other workaround. These experimental flags have just been enabled on our code basis and are not in active use by developers, so deactivating them does not do any great harm (yet). I really try to understand why all these files are marked as pre-release even though they don't use any of these features.

**Comment by 3flex** [comment]
Yes please try disabling all unused flags and either confirm the issue goes away or share an updated compiler config.

**Comment by BraisGabin** [comment]
> Ok, question here is whether this results into the non-analysis of all the offended files.

No, it shouldn't. But some rules could not catch some errors because of that. But it shouldn't imply automatically non-analysis on those files.

You can prove this really easy by adding an error in one of those files and running detekt.

**Comment by realdadfish** [comment]
Upstream told me to add `-Xskip-prerelease-check` to my `freeCompilerArgs`, but that did not work, Detekt still tells me that there have been compiler errors during analysis. I added this to my Detekt build conventions `build.gradle.kts` file:

```kotlin
configure<KotlinJvmProjectExtension> {
    compilerOptions {
        freeCompilerArgs.add(
            "-Xskip-prerelease-check"
        )
    }
}
```

I'm not sure though whether this is the right place or not.

**Comment by miduch** [comment]
I wanted to use "-Xcontext-parameters" 

but detekt started to give error 

> java.lang.IllegalStateException: Analyzing  .........  led to an exception.
  Running detekt '1.23.8' on Java '21.0.8+9-LTS' on OS 'Mac OS X'
  If the exception message does not help, please feel free to create an issue on our GitHub page.

**Comment by 3flex** [comment]
@miduch that has nothing to do with this issue (it's a different error) and it's already fixed in the alpha release, so please give that a go.

**Comment by hudson155** [comment]
@realdadfish did you find a solution? I'm also seeing this error and wondering if it's my compiler flags

**Comment by realdadfish** [comment]
No, haven't found a solution for this.

**Comment by hudson155** [comment]
I fixed it in Kairo by removing `-Xnested-type-aliases` 😄 hopefully it's just a matter of another compiler flag for you
https://github.com/hudson155/kairo/pull/971
